adafruit_wiznet5k/adafruit_wiznet5k.py

```diff
@@ -567,19 +567,22 @@ def socket_listen(self, socket_num, port):
             )
         # Initialize a socket and set the mode
         self.src_port = port
-        res = self.socket_open(socket_num, conn_mode=SNMR_TCP)
+        res = self.socket_open(socket_num, conn_mode=conn_mode)
         self.src_port = 0
         if res == 1:
             raise RuntimeError("Failed to initalize the socket.")
         # Send listen command
         self._send_socket_cmd(socket_num, CMD_SOCK_LISTEN)
         # Wait until ready
         status = [SNSR_SOCK_CLOSED]
-        while status[0] not in (SNSR_SOCK_LISTEN, SNSR_SOCK_ESTABLISHED):
+        while status[0] not in (SNSR_SOCK_LISTEN, SNSR_SOCK_ESTABLISHED, SNSR_SOCK_UDP):
             status = self._read_snsr(socket_num)
             if status[0] == SNSR_SOCK_CLOSED:
                 raise RuntimeError("Listening socket closed.")
 
+    def socket_listen_udp(self, socket_num, port):
```

Comment:
Introducing this function seemed the easiest approach at this moment. Happy to hear other thoughts.

Reply:
Could this code be placed inside `bind` instead of here?

Reply:
I was avoiding that because of access to the `adafruit_wiznet5k.adafruit_wiznet5k.SNMR_UDP` constant, but since then I have seen that `adafruit_wiznet5k.adafruit_wiznet5k_socket` does refer to the internal constants.

I'll make the change.